cmd/kogito/command/deploy/deploy_service.go

```diff
@@ -151,13 +164,24 @@ func (i *deployCommand) InitHook() {
 	i.command.Flags().StringVar(&i.flags.imageVersion, "image-version", "", "Image version for standard Kogito build images. Ignored if a custom image is set for image-s2i or image-runtime.")
 	i.command.Flags().StringVar(&i.flags.mavenMirrorURL, "maven-mirror-url", "", "Internal Maven Mirror to be used during source-to-image builds to considerably increase build speed, e.g: https://my.internal.nexus/content/group/public")
 	i.command.Flags().BoolVar(&i.flags.enableIstio, "enable-istio", false, "Enable Istio integration by annotating the Kogito service pods with the right value for Istio controller to inject sidecars on it. Defaults to false")
+	i.command.Flags().StringVarP(&i.flags.fromFile, "from-file", "f", "", "If set, a build from a single file, no pom.xml needed, will start, you can provide a entire directory, a single file from filesystem or url, the supported files are dmn, drl, bpmn or bpmn2. Porperties files is also allowed.")
```

Comment:
Why are we using an option and not the `[SOURCE]` argument itself for transmitting those source files ?
We could then leverage the `matchKogitoResources` on that source.
If match, then it is from source file.
If not then we consider a Git repository.
what do you think ?

Reply:
I could work, but IMHO, I think is better keep a separate flag/option for it. since we have plans to integrate with the binary builds later.

I would like to hear opnion from others about it.
@sutaakar @Sgitario @MarianMacik

Reply:
Sorry in advance if I misunderstood something: as far I can see the difference is that the [SOURCE] argument points out to a folder that contains always a Kogito project (with pom.xml) and the new "--from-file" flag is to specify the supported source files and also a folder (with these files). 

Taking into account that the supported source files will likely include also Java files in the future, I don't see much difference between [SOURCE] or "--from-file" options. And having them separated might make more complicated the maintenance:

- what to do if we specify the two options at the same time? 
- both options have the same behaviour using the same sources?) 

Therefore, unless this is very complex, I think it makes sense to do it smart enough to adapt the [SOURCE] to behave accordingly to what is found in the folder (be a kogito project, a binary, spare files, etc).

Reply:
The source for now is a Github repository as we needed it for s2i builds.
For now, the source has precedence over the --from-file option.
That is why I was proposing to extend its purpose and avoid another option.

Reply:
alright, I made some tweaks and looks like it is working, no flag needed.